src/update_bullet.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file update_bullet.c
+ * @author Mahsa
+ * @brief updates bullet position
+ */
+#include <stdio.h>
+#include <assert.h>
+#include "../include/map.h"
+#include "../include/single_player.h"
+
+ /**
+  * @brief updates the position of generated bullet
+  *
+  * This function shoot a bullet in a direction in which an oppenent exists.
+  * @param [in] map represent the map structure which has bullet's position and direction.
+  */
+void update_bullet(map_t* map) {
+
+	if (map->bullet[BULLET_1].speed != 0) { /**< We have a moving bullet shooted from player1*/
+		switch (map->bullet[BULLET_1].direction)
+		{
+		case DIRECTION_RIGHT:
+			if (map->bullet[BULLET_1].current_pos.x <= map->space.x_max)
+			{
+				map->bullet[BULLET_1].current_pos.x += map->bullet[BULLET_1].speed;
+			}
+			else {
+				map->bullet[BULLET_1].current_pos.x = 10000000;
```

Comment:
So we will have lots of bullet hiding in the map somewhere :) and they should be processed each time.
If you define is_bullet_active as one of the player attribute you can switch it to be false and does not show that in generate view.

Reply:
Yes you are right that's better solution indeed.